libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -188,18 +188,59 @@ bool IRGeneratorForStatements::visit(Return const& _return)
 	return false;
 }
 
+void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
+{
+	if (_unaryOperation.annotation().type->category() == Type::Category::RationalNumber)
```

Comment:
Perhaps declare a local variable `auto const* type = _unaryOperation.annotation().type`? But it also fine like this..

Reply:
Ah, and maybe  `solAssert(_unaryOperation.annotation().type, "")`

Reply:
This is something the type checker should guarantee, but perhaps I can add a function `type(Expression const&)` that always checks it.